src/ui/win32/RichPresenceDialog.cpp

```diff
@@ -0,0 +1,113 @@
+#include "RichPresenceDialog.hh"
+
+#include "RA_Core.h"
+#include "RA_Resource.h"
+#include "RA_RichPresence.h"
+
+namespace ra {
+namespace ui {
+namespace win32 {
+
+bool RichPresenceDialog::Presenter::IsSupported(const ra::ui::WindowViewModelBase& oViewModel)
+{
+    return (dynamic_cast<const ra::ui::viewmodels::RichPresenceMonitorViewModel*>(&oViewModel) != nullptr);
+}
+
+void RichPresenceDialog::Presenter::ShowModal(ra::ui::WindowViewModelBase& oViewModel)
+{
+    ShowWindow(oViewModel);
+}
+
+void RichPresenceDialog::Presenter::ShowWindow(ra::ui::WindowViewModelBase& oViewModel)
+{
+    auto& oRichPresenceViewModel = reinterpret_cast<ra::ui::viewmodels::RichPresenceMonitorViewModel&>(oViewModel);
+
+    if (m_pDialog == nullptr)
+    {
+        m_pDialog.reset(new RichPresenceDialog(oRichPresenceViewModel));
+        m_pDialog->CreateDialogWindow(MAKEINTRESOURCE(IDD_RA_RICHPRESENCE), this);
+    }
+    
+    if (m_pDialog->ShowDialogWindow())
+        m_pDialog->StartTimer();
+}
+
+void RichPresenceDialog::Presenter::OnClosed()
+{
+    m_pDialog.reset();
+}
+
+// ------------------------------------
+
+RichPresenceDialog::RichPresenceDialog(ra::ui::viewmodels::RichPresenceMonitorViewModel& vmRichPresenceDisplay)
+    : DialogBase(vmRichPresenceDisplay)
+{
+    m_hFont = CreateFont(15, 0, 0, 0, FW_NORMAL, FALSE, FALSE, FALSE, DEFAULT_CHARSET, OUT_DEFAULT_PRECIS, CLIP_DEFAULT_PRECIS, CLEARTYPE_QUALITY, VARIABLE_PITCH, nullptr);
+
+    m_bindWindow.SetInitialPosition(ra::ui::win32::bindings::RelativePosition::After, ra::ui::win32::bindings::RelativePosition::Far, "Rich Presence Monitor");
+    m_bindWindow.BindLabel(IDC_RA_RICHPRESENCERESULTTEXT, ra::ui::viewmodels::RichPresenceMonitorViewModel::DisplayStringProperty);
+}
+
+RichPresenceDialog::~RichPresenceDialog()
+{
+    DeleteFont(m_hFont);
```

Comment:
Nothing's wrong here, but I just wanted to bring that I'll eventually finalize the RAII-style wrappers for GDI handles (`HDC `will have it's own one) so we don't have to remember to delete these.

I have a few sitting around on my computer, I already made some for `HWND `(app window, client window, modal, modeless), `HANDLE `(just for a few of the kernel objects),` HGLOBAL, HFILE, HINSTANCE/HMODULE, GDI` ones, HDC (was too complicated to put with the other `GDI `ones), `HICON`, `HINTERNET`. I think the last time I was working on it was for `HWND`, haven't touched it in a while.